sys/signal.c

```diff
@@ -0,0 +1,158 @@
+#define KL_LOG KL_SIGNAL
+#include <klog.h>
+#include <signal.h>
+#include <thread.h>
+#include <malloc.h>
+#include <stdc.h>
+#include <errno.h>
+#include <sysent.h>
+#include <proc.h>
+
+static sighandler_t *def_sigact[NSIG] = {
```

Comment:
I'll update my implementation to use this way of defining default actions, but I think it actually slightly complicates things: note that they still need to use both `SA_IGNORED` and `SIG_IGN`, [like this](http://bxr.su/FreeBSD/sys/kern/kern_sig.c#737).